src/histbuf.rs

```diff
@@ -195,26 +205,37 @@ where
     }
 }
 
+impl<T, const N: usize> Drop for HistoryBuffer<T, N> {
+    fn drop(&mut self) {
+        unsafe {
+            ptr::drop_in_place(ptr::slice_from_raw_parts_mut(
+                self.data.as_mut_ptr() as *mut T,
+                self.len(),
+            ))
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::{consts::*, HistoryBuffer};
+    use crate::HistoryBuffer;
 
     #[test]
     fn new() {
-        let x: HistoryBuffer<u8, U4> = HistoryBuffer::new_with(1);
+        let x: HistoryBuffer<u8, 4> = HistoryBuffer::new_with(1);
         assert_eq!(x.len(), 4);
```

Comment:
I thought `.len()` should be 1, `.capacity()` should be 4

Reply:
`new_with` fills the entire buffer with a specific value, so `len` will be 4.